extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolTemplate.java

```diff
@@ -13,30 +13,23 @@
 @Template
 public class PgPoolTemplate {
 
-    // Visible for testing
-    static volatile PgPool pgPool;
-
     public RuntimeValue<PgPool> configurePgPool(RuntimeValue<Vertx> vertx, BeanContainer container,
             DataSourceConfig dataSourceConfig, PgPoolConfig pgPoolConfig, LaunchMode launchMode, ShutdownContext shutdown) {
 
-        initialize(vertx.getValue(), dataSourceConfig, pgPoolConfig);
+        PgPool pgPool = initialize(vertx.getValue(), dataSourceConfig, pgPoolConfig);
 
         PgPoolProducer producer = container.instance(PgPoolProducer.class);
         producer.initialize(pgPool);
 
         if (!launchMode.isDevOrTest()) {
-            shutdown.addShutdownTask(this::destroy);
+            shutdown.addShutdownTask(pgPool::close);
         }
         return new RuntimeValue<>(pgPool);
     }
 
-    // Visible for testing
-    void initialize(Vertx vertx, DataSourceConfig dataSourceConfig, PgPoolConfig pgPoolConfig) {
-        if (pgPool != null) {
-            return;
-        }
+    private PgPool initialize(Vertx vertx, DataSourceConfig dataSourceConfig, PgPoolConfig pgPoolConfig) {
         PgPoolOptions pgPoolOptions = toPgPoolOptions(dataSourceConfig, pgPoolConfig);
-        pgPool = PgClient.pool(vertx, pgPoolOptions);
+        return PgClient.pool(vertx, pgPoolOptions);
```

Comment:
Don't you need to close the previous one before starting a new one? Otherwise you could starve the database resources?

(Note that I'm not familiar with this code so the comments are of general nature and might not apply in this specific case)

Reply:
I missed that, thanks!

Reply:
Should be okay now. I'm not very familiar with this code either, this is the first change I'm doing in this area.